src/utils/source.js

```diff
@@ -122,32 +122,39 @@ const contentTypeModeMap = {
  */
 
 function getMode(source: Source) {
-  if (!source.text) {
+  const { contentType, text } = source;
+
+  if (!text) {
     return { name: "text" };
   }
 
-  if (!source.contentType) {
-    // Use HTML mode for files in which the first non whitespace
-    // character is `<` regardless of extension.
-    const name = source.text.match(/^\s*</) ? "htmlmixed" : "text";
-    return { name };
+  // Use HTML mode for files in which the first non whitespace
+  // character is `<` regardless of extension.
+  const isHTMLLike = text.match(/^\s*</);
+  if (!contentType) {
+    if (isHTMLLike) {
+      return { name: "htmlmixed" };
+    }
+    return { name: "text" };
   }
 
-  const { contentType, text } = source;
-
   // // @flow or /* @flow */
   if (text.match(/^\s*(\/\/ @flow|\/\* @flow \*\/)/)) {
     return contentTypeModeMap["text/typescript"];
   }
 
-  if (/script|elm|jsx|clojure|wasm/.test(contentType)) {
+  if (/script|elm|jsx|clojure|wasm|html/.test(contentType)) {
     if (contentType in contentTypeModeMap) {
       return contentTypeModeMap[contentType];
     }
 
     return contentTypeModeMap["text/javascript"];
   }
 
+  if (isHTMLLike) {
```

Comment:
I added a second check here because it can happen that we do have a `contentType = "text/html"` or it is equal to something else. In that case, we would still like to render it as `htmlmixed`.

I believe that this logic might be simplified if in `devtools-core/devtools-source-maps` we set html contentType to `text/html` for file extensions `.html`.

In fact, I'm not sure if there's any case for `(!contentType && isHTMLLike)` as far as I could tell in any codepath.